File: Data Structures & Algorithms/two-integer-sum/submission-3.js
class Solution {
    /**
     * @param {number[]} nums
     * @param {number} target
     * @return {number[]}
     */
    twoSum(nums, target) {
        let seen = {};
	    for (let i = 0; i < nums.length; i++) {
		    const num = nums[i];
		    const remainder = target - nums[i];
		    if (seen[remainder] !== undefined) {
			    return [seen[remainder], i];
		    } else {
			    seen[num] = i;
		    }
	    }
    }
}

File: Data Structures & Algorithms/find-duplicate-integer/submission-3.py
class Solution:
    def findDuplicate(self, nums: List[int]) -> int:
        for n in nums:
            n_to_eval = nums[abs(n) - 1]
            if n_to_eval < 0:
                return abs(n)
            nums[abs(n) - 1] = n_to_eval * -1
            